evolver/hardware/interface.py

```diff
@@ -25,6 +26,18 @@ def __init__(self, *args, evolver=None, **kwargs):
         self.evolver = evolver
         super().__init__(*args, **kwargs)
 
+    def _transform(self, transformer: str, func: str, x: Any, vial: int = None):
```

Comment:
I wonder if it might be more general if we don't call the `func` but rather return a transformer, and provide a signature more like `transformer_in(self, vial) -> Transformer` (effectors) and `transformer_out(self, vial) -> Transformer` (sensors), called like:

```py
temp = self.transformer_out(vial).convert_to(raw)
```

it can be reused a bit easer since one could stash the transformer to make further calls if needed. It reads slightly more intentional (arguably)

Reply:
Could do, but I didn't necessarily want to add it to the hardware interface, which the above suggestion would. There'd then need to be two new funcs: ``transformer_in`` and ``transformer_out`` (which I think would be better suited as properties), however, this would still leave most of the boilerplate as they could both return ``None`` and the caller would still have to check for this before calling, e.g., ``transformer.convert_to``, and handle the noop.  So it would read:
```python
if transformer:=self.transform_in(vial):
  temp = transformer.convert_to(raw)
else:
  temp = raw
```
or
```python
temp = self.transformer_in(vial).convert_to(raw) if self.transformer_in(vial) else raw
```
Thoughts?